tests/test_pagure.py

```diff
@@ -0,0 +1,749 @@
+        """Test whether the search_fields is properly set"""
+
+        body = read_file('data/pagure/pagure_repo_issue_1')
+
+        httpretty.register_uri(httpretty.GET,
+                               PAGURE_ISSUES_URL,
+                               body=body,
+                               status=200,
+                               )
+
+        pagure = Pagure(repository='Project-example', api_token='aaa')
+        issues = [issues for issues in pagure.fetch(from_date=None, to_date=None)]
+
+        issue = issues[0]
+        self.assertEqual(pagure.metadata_id(issue['data']), issue['search_fields']['item_id'])
+        self.assertIsNone(issue['search_fields']['namespace'])
+        self.assertEqual(issue['search_fields']['repo'], 'Project-example')
+
+    @httpretty.activate
+    def test_fetch_issues_reaction(self):
+        """Test the reactions on an issue comment"""
+
+        issue_2 = read_file('data/pagure/pagure_repo_issue_2')
+
+        httpretty.register_uri(httpretty.GET,
+                               PAGURE_ISSUES_URL,
+                               body=issue_2,
+                               status=200,
+                               )
+
+        pagure = Pagure(repository='Project-example')
+        issues = [issues for issues in pagure.fetch()]
+
+        self.assertEqual(len(issues), 2)
+
+        issue = issues[0]
+        self.assertEqual(issue['origin'], 'https://pagure.io/Project-example')
+        self.assertEqual(issue['uuid'], '41071b08dd75f34ca92c6d5ecb844e7a3e5939c6')
+        self.assertEqual(issue['updated_on'], 1583508642.0)
+        self.assertEqual(issue['category'], CATEGORY_ISSUE)
+        self.assertEqual(issue['tag'], 'https://pagure.io/Project-example')
+        self.assertEqual(len(issue['data']['comments']), 1)
+        self.assertEqual(issue['data']['comments'][0]['user']['name'], 'animeshk08')
+        self.assertEqual(issue['data']['assignee']['name'], 'animeshk08')
+        self.assertEqual(len(issue['data']['comments'][0]['reactions']), 0)
+
+        issue = issues[1]
+        self.assertEqual(issue['origin'], 'https://pagure.io/Project-example')
+        self.assertEqual(issue['uuid'], '7dd3642664c8a7e475814b9037277df775657850')
+        self.assertEqual(issue['updated_on'], 1583558174.0)
+        self.assertEqual(issue['category'], CATEGORY_ISSUE)
+        self.assertEqual(issue['tag'], 'https://pagure.io/Project-example')
+        self.assertEqual(issue['data']['assignee']['name'], 'animeshk0806')
+        self.assertEqual(len(issue['data']['comments']), 2)
+        self.assertEqual(issue['data']['comments'][0]['user']['name'], 'animeshk08')
+        self.assertEqual(len(issue['data']['comments'][0]['reactions']), 0)
+        self.assertEqual(len(issue['data']['comments'][1]['reactions']), 1)
+        self.assertListEqual(issue['data']['comments'][1]['reactions']['Thumbs up'], ['animeshk0806'])
+
+    @httpretty.activate
+    def test_fetch_more_issues(self):
```

Comment:
The test should check that issues on different pages are retrieved (ex.: https://github.com/chaoss/grimoirelab-perceval/blob/master/tests/test_github.py#L1121)